sdk/storage/azure-storage-blob/azure/storage/blob/aio/_download_async.py

```diff
@@ -52,42 +61,44 @@ async def process_content(data, start_offset, end_offset, encryption):
 
 
 class _AsyncChunkDownloader(_ChunkDownloader):
-    def __init__(self, **kwargs):
+    def __init__(self, **kwargs: Any) -> None:
         super(_AsyncChunkDownloader, self).__init__(**kwargs)
-        self.stream_lock = asyncio.Lock() if kwargs.get('parallel') else None
-        self.progress_lock = asyncio.Lock() if kwargs.get('parallel') else None
+        self.stream_lock_async = asyncio.Lock() if kwargs.get('parallel') else None
+        self.progress_lock_async = asyncio.Lock() if kwargs.get('parallel') else None
 
-    async def process_chunk(self, chunk_start):
+    async def process_chunk(self, chunk_start: int) -> None:  # type: ignore [override]
         chunk_start, chunk_end = self._calculate_range(chunk_start)
         chunk_data = await self._download_chunk(chunk_start, chunk_end - 1)
         length = chunk_end - chunk_start
         if length > 0:
             await self._write_to_stream(chunk_data, chunk_start)
             await self._update_progress(length)
 
-    async def yield_chunk(self, chunk_start):
+    async def yield_chunk(self, chunk_start: int) -> bytes:  # type: ignore [override]
         chunk_start, chunk_end = self._calculate_range(chunk_start)
         return await self._download_chunk(chunk_start, chunk_end - 1)
 
-    async def _update_progress(self, length):
-        if self.progress_lock:
-            async with self.progress_lock:  # pylint: disable=not-async-context-manager
+    async def _update_progress(self, length: int) -> None:  # type: ignore [override]
+        if self.progress_lock_async:
+            async with self.progress_lock_async:  # pylint: disable=not-async-context-manager
                 self.progress_total += length
         else:
             self.progress_total += length
 
         if self.progress_hook:
-            await self.progress_hook(self.progress_total, self.total_size)
+            await cast(Callable[[int, Optional[int]], Awaitable[Any]], self.progress_hook)(self.progress_total, self.total_size)  # pylint: disable=line-too-long
```

Comment:
So I ended up reverting our changes in `_download.py` to include both the sync and async versions because.. it ended up not working anyways..

Ex:
```
await self.progress_hook(self.progress_total, self.total_size)
_download_async.py:89: error: Incompatible types in "await" (actual type "Optional[Awaitable[Any]]", expected type "Awaitable[Any]")  [misc]
```
I don't understand why it thinks it is optional.. Especially since above this is a explicit check (I tried changing it to `if self.progress_hook is not None`, did not work 😢